src/natcap/invest/stormwater.py

```diff
@@ -498,10 +500,8 @@ def execute(args):
     # up with their indices in the array. this is more efficient than
     # decrementing the whole soil group array by 1.
     retention_ratio_array = numpy.array([
-        [1 - biophysical_dict[lucode][f'rc_{soil_group}']
-            for soil_group in ['a', 'b', 'c', 'd']
-         ] for lucode in sorted_lucodes
-    ], dtype=numpy.float32)
+        1 - biophysical_df[f'rc_{soil_group}'].to_numpy()
+        for soil_group in ['a', 'b', 'c', 'd']], dtype=numpy.float32).T
```

Comment:
Sorry, this is confusing! But the behavior shouldn't be changed. The transpose is needed because I replaced the nested dictionary with a dataframe, and that effectively swaps the rows and columns.

The original code
```
numpy.array([
  [1 - biophysical_dict[lucode][f'rc_{soil_group}']
      for soil_group in ['a', 'b', 'c', 'd']
   ] for lucode in sorted_lucodes])
```
produces an array of arrays, where the rows represent lucodes and the columns represent soil groups. 

The new code
```
numpy.array([
  1 - biophysical_df[f'rc_{soil_group}'].to_numpy()
     for soil_group in ['a', 'b', 'c', 'd']
])
```
produces an array of arrays where the rows are soil groups and the columns are lucodes. So the `.T` is needed to get it back to the original format.